File: src/main/java/com/github/nsc/de/shake/parser/node/logical/LogicalAndNode.java
package com.github.nsc.de.shake.parser.node.logical;

import com.github.nsc.de.shake.lexer.characterinput.position.PositionMap;
import com.github.nsc.de.shake.parser.node.ValuedNode;

public class LogicalAndNode extends LogicalConcatenationNode {
    public LogicalAndNode(PositionMap map, ValuedNode left, ValuedNode right) {
        super(map, left, right);
    }

    @Override
    public String getOperator() {
        return "&&";
    }
}

File: parser/src/commonMain/kotlin/io/github/shakelang/shake/parser/node/logical/LogicalAndNode.kt
package io.github.shakelang.shake.parser.node.logical

import io.github.shakelang.shake.parser.node.ValuedNode
import io.github.shakelang.parseutils.characters.position.PositionMap

class LogicalAndNode(map: PositionMap, left: ValuedNode, right: ValuedNode) :
    LogicalConcatenationNode(map, left, right) {
    override val operator: String
        get() = "&&"

    override fun toJson(): Map<String, *> = mapOf("name" to "LogicalAndNode", "left" to left, "right" to right)
}